swim/ping_request_sender.go

```diff
@@ -112,6 +115,29 @@ func (p *pingRequestSender) MakeCall(ctx json.Context, res *pingResponse) <-chan
 	return errC
 }
 
+// indirectPing is used to check if a target node can be reached indirectly.
+// The indirectPing is performed by sending a specifiable amount of ping
+// requests nodes in n's membership.
+func indirectPing(n *Node, target string, amount int, timeout time.Duration) (reached bool, errs []error) {
+	resCh := sendPingRequests(n, target, amount, timeout)
+
+	// wait for responses from the ping-reqs
+	for result := range resCh {
+		switch res := result.(type) {
+		case *pingResponse:
+			if res.Ok {
+				return true, errs
+			}
+			// If the ping to the target was not-ok we want to wait for more results.
+
+		case error:
+			errs = append(errs, res)
```

Comment:
Should we have default case with a panic here, or perhaps use an anonymous struct that we treat as a union?